network/transport/v2/protocol.proto

```diff
@@ -27,13 +27,89 @@ service Protocol {
 }
 
 message Envelope {
+  bytes conversationID = 1;
+
   oneof Message {
-    Hello hello = 1000;
-    TransactionPayloadQuery transactionPayloadQuery = 100;
-    TransactionPayload transactionPayload = 101;
+    // broadcast, no conversationID
+    Gossip gossip = 101;
+    Diagnostics diagnosticsBroadcast = 102;
+
+    // request, contains a new conversationID
+    State state = 201;
+    TransactionListQuery transactionListQuery = 202;
+    TransactionRangeQuery transactionRangeQuery = 203;
+    TransactionPayloadQuery transactionPayloadQuery = 204;
+
+    // response, contains conversationID from request
+    TransactionSet transactionSet = 301;
+    TransactionList transactionList = 302;
+    TransactionPayload transactionPayload = 304;
+
+    HelloMessage hello = 1000;
   }
 }
 
+message HelloMessage {
+}
+
+// Transaction represents a transaction on the DAG.
+message Transaction {
+  // hash contains the reference of the transaction, as specified by RFC004.
+  bytes hash = 1;
+  // data contains the data of the transaction, which is a JWS as specified by RFC004.
+  bytes data = 2;
+  // payload contains the payload when it may be attached but wasn't so in the transaction already
+  optional bytes payload = 3;
+}
+
+// Gossip is a message broadcast to inform peers of the node's DAG state and recent additions to it.
+message Gossip {
```

Comment:
`State` does not contain a `conversationID`. I've moved this back in to the message to avoid confusion.
The messages should handles differently, so reusing is probably not a good idea.